### PR TITLE
web3j-cli unit test generation addition.

### PR DESCRIPTION
**Pull request by AlexandrouR**
Unit test generation on the CLI.
This PR will not pass the CI until the equivalent PR is merged into the base repo.

### CODE REVIEW

File: src/main/java/org/web3j/console/project/InteractiveOptions.java [hunk 1]
```diff
@@ -53,18 +44,57 @@ protected final String getPackageName() {
...
+    }
+
+    static Optional<String> setGeneratedTestLocation() {
+        print("Where would you like to save your tests.");
+        String outputPath = getUserInput();
+        return outputPath.isEmpty() ? Optional.empty() : Optional.of(outputPath);
+    }
+
+    static boolean userWantsTests() {
+        print("Would you like to generate unit test for your solidity contracts [Y/n] ? ");
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/InteractiveOptions.java]: Default to yes

Comment (AlexandrouR) [reply src/main/java/org/web3j/console/project/InteractiveOptions.java]: Trim white space

File: src/main/java/org/web3j/console/project/InteractiveOptions.java [hunk 2]
```diff
@@ -53,18 +44,57 @@ protected final String getPackageName() {
...
         return scanner.nextLine();
     }
 
-    private void print(final String text) {
+    private static void print(final String text) {
         System.out.println(text);
     }
+
+    static boolean overrideExistingProject() {
+        print("Looks like the project exists. Would you like to override it [Y/n] ?");
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/InteractiveOptions.java]: Default to no.

File: src/main/java/org/web3j/console/project/ProjectCreator.java [hunk 3]
```diff
@@ -76,19 +80,43 @@ public static void main(String[] args) {
...
                     .build();
-            exitSuccess(
-                    "Project created with name: "
-                            + projectStructure.getProjectName()
-                            + " at location: "
-                            + projectStructure.getProjectRoot());
+            generateTests();
+            onSuccess();
         } catch (final Exception e) {
+            e.printStackTrace();
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/ProjectCreator.java]: remove print stack trace

File: src/main/java/org/web3j/console/project/UnitTestCLIRunner.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+package org.web3j.console.project;
+
+import java.io.IOException;
+
+import picocli.CommandLine;
+
+import org.web3j.codegen.Console;
+
+@CommandLine.Command(
+        name = "generate",
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/UnitTestCLIRunner.java]: this should be generate-tests

File: src/main/java/org/web3j/console/project/ProjectImporterCLIRunner.java [hunk 5]
```diff
@@ -12,34 +12,64 @@
...
 
     @Option(
             names = {"-s", "--solidity-path"},
-            description = "path to solidity file/folder",
+            description = "Path to solidity file/folder",
             required = true)
     String solidityImportPath;
 
+    @Option(
+            names = {"-t", "--generate-unit-tests"},
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/ProjectImporterCLIRunner.java]: Change to generate-tests

File: src/main/java/org/web3j/console/project/UnitTestCLIRunner.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        mixinStandardHelpOptions = true,
+        version = "4.0",
+        sortOptions = false)
+public class UnitTestCLIRunner implements Runnable {
+    @CommandLine.Option(
+            names = {"-i", "--java-wrapper-directory"},
+            description = "the class path of your generated wrapper",
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/UnitTestCLIRunner.java]: Capital T

File: src/main/java/org/web3j/console/project/UnitTestCLIRunner.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            required = true)
+    String javaWrapperDir;
+
+    @CommandLine.Option(
+            names = {"-o", "--output-directory"},
+            description = "the path where the unit tests will be generated",
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/UnitTestCLIRunner.java]: Capital T

File: src/main/java/org/web3j/console/project/UnitTestCreator.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+import picocli.CommandLine;
+
+import org.web3j.codegen.Console;
+import org.web3j.codegen.unit.gen.ClassProvider;
+import org.web3j.codegen.unit.gen.UnitClassGenerator;
+
+import static org.web3j.utils.Collection.tail;
+
+public class UnitTestCreator {
+    public static final String COMMAND_GENERATE_TESTS = "generateTests";
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/UnitTestCreator.java]: change to generate-tests

File: src/main/java/org/web3j/console/project/utills/InputVerifier.java [hunk 9]
```diff
@@ -51,4 +52,11 @@ public static String capitalizeFirstLetter(String input) {
         }
         return input.substring(0, 1).toUpperCase() + input.substring(1);
     }
+
```
Comment (AlexandrouR) [inline src/main/java/org/web3j/console/project/utills/InputVerifier.java]: simply this.

File: src/main/resources/build.gradle.template [hunk 10]
```diff
@@ -29,5 +32,10 @@ dependencies {
     implementation "org.web3j:core:$web3jVersion",
             "ch.qos.logback:logback-core:$logbackVersion",
             "ch.qos.logback:logback-classic:$logbackVersion"
-    testImplementation "junit:junit:$junitVersion"
-}
\ No newline at end of file
+    implementation("org.web3j:web3j-unit:4.6.0-SNAPSHOT")
```
Comment (AlexandrouR) [inline src/main/resources/build.gradle.template]: make this run by default with java 11

File: src/test/java/org/web3j/console/project/InteractiveOptionsTest.java [hunk 11]
```diff
@@ -12,19 +12,20 @@
  */
 package org.web3j.console.project;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.InputStream;
-import java.io.PrintStream;
+import java.io.*;
```
Comment (AlexandrouR) [inline src/test/java/org/web3j/console/project/InteractiveOptionsTest.java]: change star import.

File: src/test/java/org/web3j/console/project/ProjectCreatorTest.java [hunk 12]
```diff
@@ -31,8 +30,8 @@
 
 import org.web3j.console.project.utills.ClassExecutor;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import static java.io.File.separator;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (AlexandrouR) [inline src/test/java/org/web3j/console/project/ProjectCreatorTest.java]: change star import.

File: src/main/java/org/web3j/console/project/utills/ProjectUtils.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+            Files.walk(directoryToDeleted)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(ProjectUtils::deleteHelper);
+        } catch (IOException e) {
+            Console.exitError(e);
+        }
+    }
+
+    static void deleteHelper(File fileToDelete) {
```
Comment (Puneetha17) [inline src/main/java/org/web3j/console/project/utills/ProjectUtils.java]:
```suggestion
    static void deleteFile(File fileToDelete) {
```

File: src/main/java/org/web3j/console/project/ProjectCreatorCLIRunner.java [hunk 14]
```diff
@@ -20,40 +21,62 @@
...
-            try {
-                new ProjectCreator(outputDir, packageName, projectName).generate();
-            } catch (final IOException e) {
-                exitError(e);
+        if (inputIsValid()) {
+            if (InputVerifier.projectExists(new File(projectName))) {
+                if (overrideExistingProject()) {
+                    deleteFolder(new File(projectName).toPath());
+                    createProject();
+
```
Comment (Puneetha17) [inline src/main/java/org/web3j/console/project/ProjectCreatorCLIRunner.java]: This file has many unnecessary new lines.

**Pull request merged**